vsphere/internal/helper/storagepod/storage_pod_helper.go

```diff
@@ -305,16 +335,16 @@ func RelocateVM(
 		Type:         string(types.StoragePlacementSpecPlacementTypeRelocate),
 	}
 
-	_, err = recommendAndApplySDRS(client, sps, time.Minute*time.Duration(timeout))
+	placement, err := recommendSDRS(client, sps, time.Minute*time.Duration(timeout))
+	if err != nil {
+		return err
+	}
+	_, err = applySDRS(client, placement, time.Minute*time.Duration(timeout))
 	return err
 }
 
-func recommendAndApplySDRS(
```

Comment:
Rather than deleting this function, I'd recommend taking the de-coupled logic that has been duplicated to other parts and just throwing that here. Then the rest of the places that reference `recommendAndApplySDRS` that don't require a special path to address vApp containers don't need changes at all.

Reply:
The reason I did it this way was because I still need the recommendation in order to determine which datastore to use in the case where the VM is being created in a vApp. I didn't want to clutter up recommendAndApplySDRS with the vApp code. Does that make sense? Very possible I'm just overlooking something simple.

Reply:
I'm not saying to actually revert the de-coupling, but rather than removing the function altogether and then duplicating code a few places, just change recommendAndApplySDRS to:

```go
func recommendAndApplySDRS(
	client *govmomi.Client,
	sps types.StoragePlacementSpec,
	timeout time.Duration,
) (*object.VirtualMachine, error) {
	placement, err := recommendSDRS(client, sps, time.Minute*time.Duration(timeout))
	if err != nil {
		return nil, err
	}
	return applySDRS(client, placement, time.Minute*time.Duration(timeout))
}
```

This way, you don't have to update the other places that this is being used (where you don't need the recommendation), such as https://github.com/terraform-providers/terraform-provider-vsphere/pull/566#discussion_r202789715, https://github.com/terraform-providers/terraform-provider-vsphere/pull/566#discussion_r202789732, or https://github.com/terraform-providers/terraform-provider-vsphere/pull/566#discussion_r202789745.